src/common/stream_sync.rs

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (C) 2022 Vaticle
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+pub use std::iter::Iterator as Stream;
+
+use tokio::sync::mpsc::UnboundedReceiver;
+
+pub type BoxStream<'a, T> = Box<dyn Stream<Item = T> + Send + 'a>;
```

Comment:
Does it make sense for `BoxStream<...Stream<...>>` to be something akin to an 'Interface' in Java, ie. a Trait in Rust?

Reply:
No, we need to be able to own it and pass it around, which makes it a type. `Stream` and `Iterator` are traits though, so we are kind of passing them around like interfaces, it's just that the `Box<_>` provides extra ownership guarantees.